classes/test/NioFilesPaths.java

```diff
@@ -0,0 +1,179 @@
+      System.out.println("Does tempDir exist now?: " + Files.exists(p));
+      System.out.println("Deleting tempDir: " + Files.deleteIfExists(p));
+      System.out.println("Does tempDir exist now?: " + Files.exists(p));
+    }
+    {
+      Path p = Paths.get("/tmp", "tempDir/tempDir");
+      Path p2 = Paths.get("/tmp", "tempDir");
+      System.out.println("Does tempDir/tempDir exist?: " + Files.exists(p));
+      System.out.println("Making tempDir/tempDir : " + Files.createDirectories(p));
+      System.out.println("Does tempDir/tempDir exist now?: " + Files.exists(p));
+      System.out.println("Does tempDir exist now?: " + Files.exists(p2));
+      try {
+        System.out.println("Deleting tempDir (should fail -- nonempty): " + Files.deleteIfExists(p2));
+      } catch (final DirectoryNotEmptyException dne) {
+        System.out.println("Couldn't delete non-empty directory");
+      }
+      System.out.println("Deleting tempDir/tempDir: " + Files.deleteIfExists(p));
+      System.out.println("Deleting tempDir: " + Files.deleteIfExists(p2));
+      System.out.println("Does tempDir/tempDir exist now?: " + Files.exists(p));
+      System.out.println("Does tempDir exist now?: " + Files.exists(p2));
+    }
+
+    {
+      final Path p = Paths.get("/tmp");
+      try {
+        System.out.println("Trying to create a directory that already exists: " + Files.createDirectory(p));
+      } catch (final FileAlreadyExistsException fae) {
+        System.out.println("couldn't create a directory that already exists");
+      }
+    }
+
+    /*
+    // Rename a file.
+    {
+      File f = new File("/tmp/temp_rename_file.txt");
+      File f2 = new File("/tmp/temp_rename_file2.txt");
+      System.out.println("Creating temp_rename_file.txt: " + f.createNewFile());
+      System.out.println("Renaming it to temp_rename_file2.txt: " + f.renameTo(f2));
+      System.out.println("Old file exist? " + f.exists() + " New file exists? " + f2.exists());
+      System.out.println("Recreating old file: " + f.createNewFile());
+      System.out.println("Moving on top of old file: " + f2.renameTo(f));
+      System.out.println("Deleting old file: " + f.delete());
+      System.out.println("Trying to move nonexistant old file: " + f.renameTo(f2));
+    }
+
+    // Read only.
+    {
+      File f = new File("/tmp/temp_readonly.txt");
+      System.out.println("Creating temp_readonly.txt: " + f.createNewFile());
+      // BrowserFS doesn't support chmod in its temporary file system anymore.
+      // System.out.println("Marking as read only: " + f.setReadOnly());
+      System.out.println("Can I write to the file?: " + f.canWrite());
+      System.out.println("Can I read the file?: " + f.canRead());
+      // make sure we can open a read-only file with RandomAccessFile
+      RandomAccessFile raf = new RandomAccessFile(f, "r");
+      System.out.println("Deleting file: " + f.delete());
+    }
+    */
```

Comment:
I left the above commented tests as TODOs. I should have added a TODO comment there; will do so.

Else, if you want me to simply remove them, I can do that.